openhtf/io/output/mfg_inspector.py

```diff
@@ -439,9 +448,31 @@ def __call__(self, test_record):  # pylint: disable=invalid-name
     credentials.set_store(self._MemStorage())
 
     testrun = _TestRunFromTestRecord(test_record)
+    self.UploadTestRun(testrun, self.DESTINATION_URL, credentials)
+
+class UploadOrOutput(object):
+  """Attempt to upload to inspector, output to local if fail.
+
+  Args:
+    user: Google cloud service account for Oauth2client.
+    keydata: Google cloud key data for Oauth2client.
+    filename_pattern: A format string specifying the filename to write to,
+      will be formatted with the Test Record as a dictionary.  May also be a
+      file-like object to write directly to.
+    error_message: Message to display on upload error.
+  """
+
+  def __init__(self, user, keydata, filename_pattern, error_message=ERROR_MESSAGE):
+    self._error_message = error_message
+    self._UploadToMfgInspector = UploadToMfgInspector(user, keydata)
+    self._OutputToTestRunProto = OutputToTestRunProto(filename_pattern)
+
+  def __call__(self, test_record):  # pylint: disable=invalid-name
     try:
-      self.UploadTestRun(testrun, self.DESTINATION_URL, credentials)
-    except UploadFailedError:
-      # For now, just log the exception.  Once output is a bit more robust,
-      # we can propagate this up and handle it accordingly.
-      logging.exception('Upload to mfg-inspector failed!')
+      sys.stdout.write('Attempting to upload to Mfg Inspector...\n')
+      self._UploadToMfgInspector(test_record)
+    except Exception:
+      sys.stdout.write('%s\n' % self._error_message)
```

Comment:
by default, yes, but that's configurable by users.  the first message isn't really a warning, and so should be info, the error one could be warning or error.